public/index.html

```diff
@@ -74,22 +74,52 @@
     simulation.force("link")
         .links(graph.links);
 
-    var requestsUsedEl = document.getElementById("requests_used");
-    requestsUsedEl.textContent = graph.requestsUsed + "/" + graph.maxRequests;
+    function ticked() {
+      link
+          .attr("x1", function(d) { return d.source.x; })
+          .attr("y1", function(d) { return d.source.y; })
+          .attr("x2", function(d) { return d.target.x; })
+          .attr("y2", function(d) { return d.target.y; });
 
-    var nodesNumberEl = document.getElementById("nodes_number");
-    nodesNumberEl.textContent = graph.nodes.length;
+      node
+          .attr("cx", function(d) { return d.x = Math.max(5, Math.min(width - 5, d.x)); })
+          .attr("cy", function(d) { return d.y = Math.max(5, Math.min(height - 5, d.y)); });
+          //.attr("cx", function(d) { return d.x; })
+          //.attr("cy", function(d) { return d.y; });
+    }
+  });
+
+  d3.json("/dashboarddata.json", function(error, dashboard) {
 
-    var lastUpdateEl = document.getElementById("last_update");
-    lastUpdateEl.textContent = graph.lastUpdate;
+    var statusEl = document.getElementById("status");
+    if (error) {
+      statusEl.textContent = "Content currently unavailable. Please check server logs.";
+      return;
+    }
+
+    statusEl.textContent = "All operational";
+
+    var requestsUsedEl = document.getElementById("requests_used");
+    requestsUsedEl.textContent = dashboard.requestsUsed + "/" + dashboard.maxRequests;
 
     var refreshIntervalEl = document.getElementById("refresh_interval");
 
+    var lastUpdate;
+
+    d3.json("/graphdata.json", function(error, graph) {
+      var nodesNumberEl = document.getElementById("nodes_number");
+      nodesNumberEl.textContent = graph.nodes.length;
+
+      var lastUpdateEl = document.getElementById("last_update");
+      lastUpdateEl.textContent = graph.lastUpdate;
+      lastUpdate = graph.lastUpdate
+    });
+
     window.setInterval(function() {
       var now = new Date();
```

Comment:
Does this have a single use instance? Is it worth allocating memory as a variable?